adc/microphone_adc/README.adoc

```diff
@@ -0,0 +1,49 @@
+= Attaching a microphone using the ADC
+
+This example code shows how to interface the Raspberry Pi Pico with a standard analog microphone via the onboard analog to digital converter (ADC). In this example, we use an ICS-40180 breakout board by SparkFun but any analog microphone should be compatible with this tutorial. SparkFun have https://learn.sparkfun.com/tutorials/mems-microphone-hookup-guide[written a guide] for this board that goes into more detail about the board and how it works.
+
+[TIP]
+======
+An analog to digital converter (ADC) is responsible for reading continuous input signals that may range from 0 to a specified reference voltage (in the Pico's case this reference voltage is set by the supply voltage and can be measured on pin 35, ADC_VREF) and converting them into 1s and 0s, ie. a number that can be digitally stored.
+======
+
+The Pico has a 12-bit ADC, meaning that a read operation will return a number ranging from 0 to 4095 (2^12 - 1) for a total of 4096 possible values. Therefore, the resolution of the ADC is 3.3/4096, so roughly steps of 0.8 millivolts. The SparkFun breakout uses an OPA344 operational amplifier to boost the signal coming from the microphone to voltage levels that can be easily read by the ADC. An important side effect is that a bias of 0.5*Vcc is added to the signal, even when the microphone is not picking up any sound.
```

Comment:
Is `Vcc` the same thing as `ADC_VREF`?

@aallan Do we need to mention ENOB here, or is it fine to pretend that the ADC is perfect?

Reply:
> @aallan Do we need to mention ENOB here, or is it fine to pretend that the ADC is perfect?

Urgh. We should probably say "The Pico has a 12-bit ADC (ENOB of 8.7-bit, see the [RP2040 Datasheet]() for details), meaning that…" with a URL that links into that specific section.